source/common/quic/envoy_quic_utils.cc

```diff
@@ -201,8 +214,18 @@ createConnectionSocket(const Network::Address::InstanceConstSharedPtr& peer_addr
     ENVOY_LOG_MISC(error, "Fail to apply pre-bind options");
     return connection_socket;
   }
-  connection_socket->bind(local_addr);
-  ASSERT(local_addr->ip());
+
+  if (local_addr != nullptr) {
+    connection_socket->bind(local_addr);
+    ASSERT(local_addr->ip());
+  }
+  if (auto result = connection_socket->connect(peer_addr); result.return_value_ == -1) {
```

Comment:
Is this method client only, or do we use it in servers too?

Reply:
this method is only called by EnvoyQuicClientConnection, so it is only called for client connections.